src/transports/ws.rs

```diff
@@ -36,20 +40,85 @@ type BatchResult = error::Result<Vec<SingleResult>>;
 type Pending = oneshot::Sender<BatchResult>;
 type Subscription = mpsc::UnboundedSender<rpc::Value>;
 
+/// Stream, either plain TCP or TLS.
+enum MaybeTlsStream<S> {
+    /// Unencrypted socket stream.
+    Plain(S),
```

Comment:
We could use `pin_project` to avoid the `Unpin` bound, but it's already a separate issue to work on. See #361